builtin/providers/librato/resource_librato_space_chart.go

```diff
@@ -0,0 +1,408 @@
+							Optional: true,
+						},
+						"max": &schema.Schema{
+							Type:     schema.TypeFloat,
+							Optional: true,
+						},
+						"transform_function": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"period": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+					},
+				},
+				Set: resourceLibratoSpaceChartHash,
+			},
+		},
+	}
+}
+
+func resourceLibratoSpaceChartHash(v interface{}) int {
+	var buf bytes.Buffer
+	m := v.(map[string]interface{})
+	buf.WriteString(fmt.Sprintf("%s-", m["metric"].(string)))
+	buf.WriteString(fmt.Sprintf("%s-", m["source"].(string)))
+	buf.WriteString(fmt.Sprintf("%s-", m["composite"].(string)))
+
+	return hashcode.String(buf.String())
+}
+
+func resourceLibratoSpaceChartCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*librato.Client)
+
+	spaceID := uint(d.Get("space_id").(int))
+
+	spaceChart := new(librato.SpaceChart)
+	if v, ok := d.GetOk("name"); ok {
+		spaceChart.Name = librato.String(v.(string))
+	}
+	if v, ok := d.GetOk("type"); ok {
+		spaceChart.Type = librato.String(v.(string))
+	}
+	if v, ok := d.GetOk("min"); ok {
+		spaceChart.Min = librato.Float(v.(float64))
+	}
+	if v, ok := d.GetOk("max"); ok {
+		spaceChart.Max = librato.Float(v.(float64))
+	}
+	if v, ok := d.GetOk("label"); ok {
+		spaceChart.Label = librato.String(v.(string))
+	}
+	if v, ok := d.GetOk("related_space"); ok {
+		spaceChart.RelatedSpace = librato.Uint(uint(v.(int)))
+	}
+	if v, ok := d.GetOk("stream"); ok {
+		vs := v.(*schema.Set)
+		streams := make([]librato.SpaceChartStream, vs.Len())
+		for i, streamDataM := range vs.List() {
+			streamData := streamDataM.(map[string]interface{})
+			var stream librato.SpaceChartStream
+			if v, ok := streamData["metric"].(string); ok && v != "" {
+				stream.Metric = librato.String(v)
+			}
+			if v, ok := streamData["source"].(string); ok && v != "" {
+				stream.Source = librato.String(v)
+			}
+			if v, ok := streamData["composite"].(string); ok && v != "" {
+				stream.Composite = librato.String(v)
+			}
+			if v, ok := streamData["group_function"].(string); ok && v != "" {
+				stream.GroupFunction = librato.String(v)
+			}
+			if v, ok := streamData["summary_function"].(string); ok && v != "" {
+				stream.SummaryFunction = librato.String(v)
+			}
+			if v, ok := streamData["transform_function"].(string); ok && v != "" {
+				stream.TransformFunction = librato.String(v)
+			}
+			if v, ok := streamData["color"].(string); ok && v != "" {
+				stream.Color = librato.String(v)
+			}
+			if v, ok := streamData["units_short"].(string); ok && v != "" {
+				stream.UnitsShort = librato.String(v)
+			}
+			if v, ok := streamData["units_longs"].(string); ok && v != "" {
+				stream.UnitsLong = librato.String(v)
+			}
+			// TODO: This isn't correct, since it means you can't set 0 as the minimum
+			if v, ok := streamData["min"].(float64); ok && v != 0 {
```

Comment:
I'm not sure how to best discover the difference between a zero value and the value not being set for nested blocks. This TODO message occurs a few other places too, but they are all essentially the same issue.